### PR TITLE
Add warning on inspecting locally-generated docs

### PR DESCRIPTION
**Pull request by ana-borges**
- ~~[ ] Added **changelog**.~~

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Comment by ana-borges** [comment]
@Zimmi48 and I ran into this problem during #17772. It led to a long and unfruitful debugging session until @maximedenes gave us this tip.

File: doc/README.md [hunk 1]
```diff
@@ -49,6 +49,15 @@ doesn't include the LaTeX packages needed to build the full documentation).
 
 You can check the dependencies using the `doc/tools/coqrst/checkdeps.py` script.
 
+Note that inspecting local copies of the docs may behave in unexpected ways if
+opening the sources with a browser (eg with `firefox
+_build/default/doc/refman-html/index.html`). In order to avoid this, either
+inspect the version generated by the CI or run a local server, for example
+with:
+```
+python3 -m http.server
```
Comment (Zimmi48) [inline doc/README.md:58]:
```suggestion
(cd _build/default/doc/refman-html/ && python3 -m http.server)
```

**Review by Zimmi48 — COMMENTED** [review]
Wouldn't it make more sense to move this piece of advice to the Compilation section (instead of the Dependencies section as it is now)?

**Comment by jfehrle** [comment]
> the Sphinx documentation framework is actually not designed in a way that allows users to install the HTML documentation for local browsing

Perhaps one of you would file a Sphinx bug report?  Better too  eventually get a fix rather than just  documenting a bug.

**Comment by Zimmi48** [comment]
> > the Sphinx documentation framework is actually not designed in a way that allows users to install the HTML documentation for local browsing
> 
> Perhaps one of you would file a Sphinx bug report? Better to eventually get a fix rather than just documenting a bug.

I don't know if that should be considered a bug, and in any case, I'm not willing to spend the required time producing a minimal example to open one. Maybe @cpitclaudel has an opinion about it, or even knows what the official Sphinx position is about this.

**Review by Zimmi48 — APPROVED** [review]
@coqbot merge now

**Pull request merged**